src/sherpa_ai/memory/vector_storage.py

```diff
@@ -0,0 +1,183 @@
+from loguru import logger
+
+import sherpa_ai.config as cfg
+from sherpa_ai.utils import load_files
+
+
+class ConversationStore(VectorStore):
+    def __init__(self, namespace, db, embeddings, text_key):
+        self.db = db
+        self.namespace = namespace
+        self.embeddings = embeddings
+        self.text_key = text_key
+
+    @classmethod
+    def from_index(cls, namespace, openai_api_key, index_name, text_key="text"):
+        pinecone.init(api_key=cfg.PINECONE_API_KEY, environment=cfg.PINECONE_ENV)
+        logger.info(f"Loading index {index_name} from Pinecone")
+        index = pinecone.Index(index_name)
+        embeddings = OpenAIEmbeddings(openai_api_key=openai_api_key)
+        return cls(namespace, index, embeddings, text_key)
+
+    def add_text(self, text: str, metadata={}) -> str:
+        metadata[self.text_key] = text
+        id = str(uuid.uuid4())
+        embedding = self.embeddings.embed_query(text)
+        doc = {"id": id, "values": embedding, "metadata": metadata}
+        self.db.upsert(vectors=[doc], namespace=self.namespace)
+
+        return id
+
+    def add_texts(self, texts: Iterable[str], metadatas: List[dict]) -> List[str]:
+        for text, metadata in zip(texts, metadatas):
+            self.add_text(text, metadata)
+
+    def similarity_search(
+        self,
+        text: str,
+        top_k: int = 5,
+        filter: Optional[dict] = None,
+        threshold: float = 0.7,
+    ) -> list[Document]:
+        query_embedding = self.embeddings.embed_query(text)
+        results = self.db.query(
+            [query_embedding],
+            top_k=top_k,
+            include_metadata=True,
+            namespace=self.namespace,
+            filter=filter,
+        )
+
+        docs = []
+        for res in results["matches"]:
+            metadata = res["metadata"]
+            text = metadata.pop(self.text_key)
+            if res["score"] > threshold:
+                docs.append(Document(page_content=text, metadata=metadata))
+        return docs
+
+    def _similarity_search_with_relevance_scores(
+        self,
+        query: str,
+        k: int = 4,
+        **kwargs: Any,
+    ) -> List[Tuple[Document, float]]:
+        logger.debug("query", query)
+        query_embedding = self.embeddings.embed_query(query)
+        results = self.db.query(
+            [query_embedding],
+            top_k=k,
+            include_metadata=True,
+            namespace=self.namespace,
+            filter=kwargs.get("filter", None),
+        )
+
+        docs_with_score = []
+        for res in results["matches"]:
+            metadata = res["metadata"]
+            text = metadata.pop(self.text_key)
+            docs_with_score.append(
+                (Document(page_content=text, metadata=metadata), res["score"])
+            )
+        logger.debug(docs_with_score)
+        return docs_with_score
+
+    @classmethod
+    def delete(cls, namespace, index_name):
```

Comment:
delete_index

Reply:
I didn't work on this class . should I create a separate pr for it or change the name on this pr ?

Reply:
change the name on this PR. 

general idea: while you are already touching the code it's an OK time to improve it a little, as long as you aren't getting distracted from your overall goal.